include/swift/AST/ModuleLoader.h

```diff
@@ -63,10 +63,10 @@ class DependencyTracker {
 
 /// \brief Abstract interface that loads named modules into the AST.
 class ModuleLoader {
-  DependencyTracker * const dependencyTracker;
   virtual void anchor();
 
 protected:
+  DependencyTracker * const dependencyTracker;
```

Comment:
The intent here was that clients would go through ModuleLoader's `addDependency`, which hides the null check. Does that not work here?

Reply:
I needed to extract the tracker to thread through a couple static (in the file-local sense) free functions. I could have turned them all into methods on the module loader but my general preference is to avoid object methods for implementations; don't like the implicit sharing, nor having to clutter up the class interface with them.

Reply:
There's always that war, but exposing the dependency tracker is leaking an abstraction. If we're going to expose it anyway, I don't feel like the helper on ModuleLoader pulls its weight.

(Also, one way to get around some of the implementation bit is `friend class Implementation;`, and then putting all the helpers in a file-local class.)

Reply:
Sure, it's just only leaking the abstraction down to subclasses (protected), not publicly. I could also use a protected getter? Or I guess if you feel strongly I'll move the relevant code into members or an implementation class.

Reply:
I guess I don't consider `protected` any better than `public` here. If we can't hide the existence of the DependencyTracker here, we shouldn't have different parts of the compiler act differently. (It isn't naturally something that belongs on a ModuleLoader; the only reason it's there at all is so that subclasses wouldn't have to go look for it somewhere else.)

Reply:
resolved by removal of the helper (subclasses see and use the member directly).